### PR TITLE
Seventeen day

### PR DESCRIPTION
**Pull request by hrodberaht**
https://trello.com/c/pbBtoAyn

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -4,7 +4,9 @@ module.exports = {
   "rules": {
     "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
     "react/prefer-stateless-function": [0],
-    "react/prop-types": [2, { ignore: [ 'history' ]}]
+    "react/prop-types": [2, { ignore: [ 'history' ]}],
+    "no-underscore-dangle": ["error", { "allow": ["__REDUX_DEVTOOLS_EXTENSION_COMPOSE__"] }],
+    "import/no-extraneous-dependencies": ["error", {"devDependencies": true}]
```
Comment (ar-mac) [inline .eslintrc.js:9]: It's not the final solution. https://trello.com/c/RFLBpjv1/17-eslint

File: src/components/Header/NavBar/Authorization/Authorization.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { NavLink, withRouter } from 'react-router-dom';
+
+import { logOut } from '../../../../actions';
+
+export class Authorization extends Component {
+    rendererLogin = () => {
```
Comment (ar-mac) [inline src/components/Header/NavBar/Authorization/Authorization.js:9]: eslint  should throw error due to the deeper indentation

**Review by ar-mac — COMMENTED** [review]
Apart from the comments, this looks awesome

File: src/components/Header/NavBar/Authorization/Authorization.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      const { handleLogOut, auth } = this.props;
+      if (auth) return <NavLink to="/login" onClick={handleLogOut}>Log out</NavLink>;
+
+      return <NavLink to="/login">Login</NavLink>;
+    }
+
+    render() {
+      return (
+        this.rendererLogin()
```
Comment (ar-mac) [inline src/components/Header/NavBar/Authorization/Authorization.js:18]: (Minor)

In current of this component contents of the rendererLogin could be placed in `render` method.

```
render() {
  const { handleLogOut, auth } = this.props;
  if (auth) return <NavLink to="/login" onClick={handleLogOut}>Log out</NavLink>;
  return <NavLink to="/login">Login</NavLink>;
}
```
It's quite common pattern to have multiple conditional returns in render. (especially useful when displaying loading, or info, that data is missing)

File: src/components/Header/NavBar/Authorization/Authorization.js (hunk 2 above)
Comment (hrodberaht) [reply src/components/Header/NavBar/Authorization/Authorization.js:9]: It didn't show, If I move one line then he shows me the error, I correct this.

File: .eslintrc.js (hunk 1 above)
Comment (hrodberaht) [reply .eslintrc.js:9]: I think, that now is ok https://github.com/hrodberaht/react-learning/commit/3f747bd683566c18e72c24ab034db684287a6f21

Comment (ar-mac) [reply .eslintrc.js:9]: It awesome 👍

**Review by ar-mac — APPROVED** [review]

**Pull request merged**